tools/ci/check_grep.sh

```diff
@@ -308,16 +302,103 @@ do
+	part "multiple closets"
+	if $grep -PU '"\w+" = \(\n[^)]*?/obj/structure/closet(?<type>[/\w]*),\n[^)]*?/obj/structure/closet\g{type},\n[^)]*?/area/.+\)' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found multiple identical closets on the same tile, please remove them.${NC}"
+		st=1
+	fi;
+	part "multiple grilles"
+	if $grep -PU '"\w+" = \(\n[^)]*?/obj/structure/grille(?<type>[/\w]*),\n[^)]*?/obj/structure/grille/\g{type},\n[^)]*?/area/.+\)' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found multiple identical grilles on the same tile, please remove them.${NC}"
+		st=1
+	fi;
+	part "multiple girders"
+	if $grep -PU '"\w+" = \(\n[^)]*?/obj/structure/girder(?<type>[/\w]*),\n[^)]*?/obj/structure/girder\g{type},\n[^)]*?/area/.+\)' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found multiple identical girders on the same tile, please remove them.${NC}"
+		st=1
+	fi;
+	part "multiple stairs"
+	if $grep -PU '"\w+" = \(\n[^)]*?/obj/structure/stairs/(?<type>[/\w]*),\n[^)]*?/obj/structure/stairs/\g{type},\n[^)]*?/area/.+\)' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found multiple identical stairs on the same tile, please remove them.${NC}"
+		st=1
+	fi;
+	part "door names"
+	if $grep -PU '/obj/machinery/door.*{([^}]|\n)*name = .*("|\s)(?!of|and|to)[a-z].*\n' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found door names without proper upper-casing. Please upper-case your door names.${NC}"
+		st=1
+	fi;
+	part "multiple chairs"
+	if $grep -PU '"\w+" = \(\n[^)]*?/obj/structure/chair(?<type>[/\w]*),\n[^)]*?/obj/structure/chair\g{type},\n[^)]*?/area/.+\)' $map_files;	then
+		echo
+		echo -e "${RED}ERROR: Found multiple identical chairs on the same tile, please remove them.${NC}"
+		st=1
+	fi;
+	part "to_chat sanity"
+	if $grep -P 'to_chat\((?!.*,).*\)' $code_files; then
+		echo
+		echo -e "${RED}ERROR: to_chat() missing arguments.${NC}"
+		st=1
+	fi;
+	part "timer flag sanity"
+	if $grep -P 'addtimer\((?=.*TIMER_OVERRIDE)(?!.*TIMER_UNIQUE).*\)' $code_files; then
+		echo
+		echo -e "${RED}ERROR: TIMER_OVERRIDE used without TIMER_UNIQUE.${NC}"
+		st=1
+	fi
+	part "trailing newlines"
+	if $grep -PU '[^\n]$(?!\n)' $code_files; then
+		echo
+		echo -e "${RED}ERROR: File(s) with no trailing newline detected, please add one.${NC}"
+		st=1
+	fi
+	part "docking_port varedits"
+	if $grep -PU '^/obj/docking_port/mobile.*\{\n[^}]*(width|height|dwidth|dheight)[^}]*[}]' $map_files; then
+		echo
+		echo -e "${RED}ERROR: Custom mobile docking_port sizes detected. This is done automatically and should not be varedits."
+		echo -e "\t\tPlease remove the width, height, dwidth, and dheight varedits from the docking_port.${NC}"
+		st=1
+	fi;
+else
+	echo -e "${RED}pcre2 not supported, skipping checks requiring pcre2"
+	echo -e "if you want to run these checks install ripgrep with pcre2 support.${NC}"
+fi
+
 if [ $st = 0 ]; then
     echo
-    echo -e "${GREEN}No errors found using grep!${NC}"
+    echo -e "${GREEN}No errors found using ripgrep!${NC}"
```

Comment:
Have this match the name of what program we used to run the greps with?

Reply:
```suggestion
    echo -e "${GREEN}No errors found using $grep!${NC}"
```